tests/integration/vrf_test.yaml

```diff
@@ -0,0 +1,369 @@
+description: Create the vlan device over test-bond0 with ID 100, create the bond interface test-bond0 with bonding mode balance-rr, create the vrf interface test-vrf0 with ports test-bond0, test-bond0.100, and the vrf route table id 100
+interfaces:
+- name: test-bond0
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-rr
+- name: test-bond0.100
+  type: vlan
+  state: up
+  vlan:
+    base-iface: test-bond0
+    id: 100
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - test-bond0
+    - test-bond0.100
+    route-table-id: 100
+description: Delete the vlan interface test-bond0.100, delete the bond interface test-bond0, delete the vrf interface test-vrf0
+interfaces:
+- name: test-bond0
+  type: bond
+  state: absent
+- name: test-bond0.100
+  type: vlan
+  state: absent
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Set the empty port for the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port: []
+description: Set the empty port for the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port: []
+description: Create the vrf interface test-vrf0 with vrf port eth1 and vrf route table ID 100
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    route-table-id: 100
+description: Delete the vrf interface test-vrf0
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf1, attach ports eth1 and eth2 to it, configure the vrf route table ID 101
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth1
+    - eth2
+    route-table-id: 101
+description: Delete the vrf interface test-vrf1
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf1 with vrf port eth2 and vrf route table ID 101
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth2
+    route-table-id: 101
+description: Delete the vrf interface test-vrf1
+interfaces:
+- name: test-vrf1
+  type: vrf
+  state: absent
+description: Create the vrf interface test-vrf0 with port eth2, create the vrf interface test-vrf1 with port eth1
+interfaces:
+- name: test-vrf0
+  type: vrf
+  state: up
+  vrf:
+    port:
+    - eth2
```

Comment:
Even it is hard to tell whether certain YAML is creating or modifying, but clearly you cannot create a VRF without route table ID defined. So this YAML is modifying the vrf interface, not create.